plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTableStatistics.java

```diff
@@ -0,0 +1,467 @@
+                        .buildOrThrow(),
+                ImmutableList.of(ORDERS));
+    }
+
+    @Override
+    @Test(invocationCount = 10, successPercentage = 50) // PostgreSQL can auto-analyze data before we SHOW STATS
+    public void testNotAnalyzed()
+    {
+        String tableName = "test_stats_not_analyzed";
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        computeActual(format("CREATE TABLE %s AS SELECT * FROM tpch.tiny.orders", tableName));
+        try {
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('orderkey', null, null, null, null, null, null)," +
+                            "('custkey', null, null, null, null, null, null)," +
+                            "('orderstatus', null, null, null, null, null, null)," +
+                            "('totalprice', null, null, null, null, null, null)," +
+                            "('orderdate', null, null, null, null, null, null)," +
+                            "('orderpriority', null, null, null, null, null, null)," +
+                            "('clerk', null, null, null, null, null, null)," +
+                            "('shippriority', null, null, null, null, null, null)," +
+                            "('comment', null, null, null, null, null, null)," +
+                            "(null, null, null, null, 15000, null, null)");
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Override
+    @Test
+    public void testBasic()
+    {
+        String tableName = "test_stats_orders";
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        computeActual(format("CREATE TABLE %s AS SELECT * FROM tpch.tiny.orders", tableName));
+        try {
+            gatherStats(tableName);
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('orderkey', null, 15000, 0, null, null, null)," +
+                            "('custkey', null, 1000, 0, null, null, null)," +
+                            "('orderstatus', 30000, 3, 0, null, null, null)," +
+                            "('totalprice', null, 14996, 0, null, null, null)," +
+                            "('orderdate', null, 2401, 0, null, null, null)," +
+                            "('orderpriority', 135000, 5, 0, null, null, null)," +
+                            "('clerk', 240000, 1000, 0, null, null, null)," +
+                            "('shippriority', null, 1, 0, null, null, null)," +
+                            "('comment', 735000, 14995, 0, null, null, null)," +
+                            "(null, null, null, null, 15000, null, null)");
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Override
+    @Test
+    public void testAllNulls()
```

Comment:
any specific reason `testAllNulls` isn't part of base class? Do we have connectors which lie about NDV (is it MySQL?)?

Reply:
i don't remember. I think i wrote this like 3 years ago :D

Reply:
Once the other two are also merged I'll see what else can be extracted.

Reply:
@hashhar let's wait for all the stats contribution to land, not only the ones i opened a PR for. would that work?